addons/web/static/lib/jquery.mjs.nestedSortable/jquery.mjs.nestedSortable.js

```diff
@@ -346,7 +346,9 @@
 					this.direction = intersection === 1 ? "down" : "up";
 
 					// mjs - rearrange the elements and reset timeouts and hovering state
-					if (this.options.tolerance === "pointer" || this._intersectsWithSides(item)) {
+					if ((this.options.tolerance === "pointer" || this._intersectsWithSides(item)) &&
+						(intersection === 1 || (itemElement.compareDocumentPosition(this.placeholder[0]) & Node.DOCUMENT_POSITION_PRECEDING))
```

Comment:
Could be cool to give a name that reflect what it the condition and use that variable in the if (to make make it more readable) #superPicky

Reply:
I would strongly advice to **not modify a library**! ping @odoo/rd-framework-js 

Or if not avoidable, make it crystal clear what has been changed and why.

Reply:
The library was not updated in ~10 years and has bugged behaviors. It would be better to rewrite it entirely internally but in the meantime, those fixes will allow a smoother behavior.